align/compiler/write_verilog_lef.py

```diff
@@ -235,86 +225,91 @@ def generate_lef(fp, name, values, available_block_lef,
-            if block_name in available_block_lef:
-                return block_name
-            logging.info('Generating lef for: %s %s', block_name, size)
-            fp.write("\n$PC fabric_" + name + ".py " +
-                     " -b " + block_name +
-                     " -n " + str(height) +
-                     " -r " + size)
-        except:
-            fp.write("\n$PC fabric_" + name + ".py " +
-                     " -b " + block_name +
-                     " -n " + '1'  +
-                     " -r " + str(res_unit_size))
-
-    elif name.lower().startswith('inductor') or \
-        name.lower().startswith('spiral'):
-        try:
-            size = round(values["ind"]*1E12,2)
-        except :
-            convert_to_unit(values)
-            size = '_'.join(param+str(values[param]) for param in values)
-            
-        ind_unit_size = unit_size_cap
-        height = ceil(sqrt(size / ind_unit_size))
-        block_name = name + '_' + str(size)
         if block_name in available_block_lef:
-            return block_name
-        logging.info('Generating lef for: %s %s', block_name, size)
+            return block_name, available_block_lef[block_name]
+        logging.info('Generating lef for: %s %s', name, size)
+        # return block_name, {
+        #     'primitive': name,
+        #     'nfin': unit_size_mos,
+        #     'x_cells': xval,
+        #     'y_cells': yval,
+        #     'parameters': values
+        # }
+
         fp.write("\n$PC fabric_" + name + ".py " +
-                 " -b " + block_name +
-                 " -n " + str(height) +
-                 " -r " + str(size))
-        
+                    " -b " + unit_block_name + 
+                    " -n " + str(unit_size_cap))
+
+    # elif name.lower().startswith('res_array_8'):
+    #     if 'res' in values.keys():
+    #         size = '%g'%(round(values["res"],2))
+    #     elif 'r' in values.keys():
+    #         size = '%g'%(round(values["r"],2))
+    #     else :
+    #         convert_to_unit(values)
+    #         size = '_'.join(param+str(values[param]) for param in values)
+    #     try:
+    #         #size = float(size)
+    #         res_unit_size = 30 * unit_size_cap
+    #         height = ceil(sqrt(float(size) / res_unit_size))
+    #         block_name = name + '_' + size.replace('.','p')
+    #         if block_name in available_block_lef:
+    #             return block_name, available_block_lef[block_name]
+    #         logging.info('Generating lef for: %s %s', block_name, size)
+    #         fp.write("\n$PC fabric_Res_array.py " +
+    #                  " -b " + block_name +
+    #                  " -n " + str(height) +
+    #                  " -X " + "8" +
+    #                  " -r " + size)
+    #     except:
+    #         block_name = name + '_' + size
+                
+    # elif name.lower().startswith('res'):
+    #     if 'res' in values.keys():
```

Comment:
Commenting out untested code. 

@kkunal1408 do we have any circuits using res?